core/vulkan/vk_api_timing_layer/cc/timing_layer.tmpl

```diff
@@ -219,16 +219,81 @@ VkResult vkDebugMarkerSetObjectNameEXT(
     return debug_marker_ext_supported ? next(device, pNameInfo) : VK_SUCCESS;
 }
 
-static const uint32_t NUM_EXTENSIONS = 2;
-static const VkExtensionProperties new_device_extensions[] = {
+
+namespace {
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
+
+const char* LAYER_NAME = "CPUTiming";
+const uint32_t NUM_EXTENSIONS = 1;
+const VkExtensionProperties NEW_INSTANCE_EXTENSIONS[] = {
     {VK_EXT_DEBUG_UTILS_EXTENSION_NAME, VK_EXT_DEBUG_UTILS_SPEC_VERSION},
+};
+const VkExtensionProperties NEW_DEVICE_EXTENSIONS[] = {
     {VK_EXT_DEBUG_MARKER_EXTENSION_NAME, VK_EXT_DEBUG_MARKER_SPEC_VERSION},
 };
 
+/**
+ * Enumerate extension properties for a specific layer.
+ *
+ * This should expose only the new extensions added by the layer.
+ */
+VkResult enumerateExtensionPropertiesForLayer(
+        uint32_t* pPropertyCount,
+        VkExtensionProperties* pProperties,
+        uint32_t numExtensions,
+        const VkExtensionProperties extensions[]) {
+    if (pProperties == NULL) {
+        *pPropertyCount = numExtensions;
+        return VK_SUCCESS;
+    }
+    uint32_t capacity = std::min(*pPropertyCount, numExtensions);
+    memcpy(pProperties, extensions, capacity * sizeof(VkExtensionProperties));
+    if (*pPropertyCount < numExtensions) {
+        return VK_INCOMPLETE;
+    } else {
+        *pPropertyCount = numExtensions;
+        return VK_SUCCESS;
+    }
+}
+} // end of anonymous namespace
+
 // This layer needs to add VK_EXT_debug_utils and VK_EXT_debug_marker as a supported extension.
-VkResult vkEnumerateDeviceExtensionProperties(PFN_vkEnumerateDeviceExtensionProperties next, VkPhysicalDevice physicalDevice, char const* pLayerName, uint32_t* pPropertyCount, VkExtensionProperties* pProperties) {
 
+VkResult vkEnumerateInstanceExtensionProperties(
+        PFN_vkEnumerateInstanceExtensionProperties next,
+        char const* pLayerName,
+        uint32_t* pPropertyCount,
+        VkExtensionProperties* pProperties) {
+    timer t("", "vkEnumerateInstanceExtensionProperties");
+    if (pLayerName != nullptr && strcmp(pLayerName, LAYER_NAME) == 0) {
+        return enumerateExtensionPropertiesForLayer(
+                pPropertyCount,
+                pProperties,
+                ARRAY_SIZE(NEW_INSTANCE_EXTENSIONS),
+                NEW_INSTANCE_EXTENSIONS);
+    }
+    return next(pLayerName, pPropertyCount, pProperties);
+}
+
+VkResult vkEnumerateDeviceExtensionProperties(
+        PFN_vkEnumerateDeviceExtensionProperties next,
+        VkPhysicalDevice physicalDevice,
+        char const* pLayerName,
+        uint32_t* pPropertyCount,
+        VkExtensionProperties* pProperties) {
     timer t("VkPhysicalDevice", "vkEnumerateDeviceExtensionProperties");
+
+    static const uint32_t NUM_EXTENSIONS = ARRAY_SIZE(NEW_DEVICE_EXTENSIONS);
```

Comment:
maybe we should have `NUM_INSTANCE_EXTENSIONS` and `NUM_DEVICE_EXTENSIONS`, or do `const uint32_t num_extensions = ARRAY_SIZE()` here.

Reply:
Done